### PR TITLE
Reduce the indirection through _outgoing in _request

### PR DESCRIPTION
**Pull request by paulmelnikow**
This isn't an obvious win, though it should facilitate further cleanup in `_makeRequest()`.

@Fishbowler I'm curious, currently [the documentation for the `params` passed to each command](https://github.com/IcedFrisby/IcedFrisby/blob/master/API.md#commands) calls out four supported parameters. The implementation, in most cases, actually passes through unknown parameters to `request()`. Do you think that's desirable? If we were to change the implementation so it only worked with the documented parameters, do you think that would be a breaking change?

### CODE REVIEW

File: lib/icedfrisby.js [hunk 1]
```diff
@@ -447,88 +448,10 @@ class Frisby {
     const args = Array.from(arguments)
     const method = args.shift()
     const uri = typeof args[0] === 'string' && args.shift()
-    let data = typeof args[0] === 'object' && args.shift()
+    const data = typeof args[0] === 'object' && args.shift()
     const params = typeof args[0] === 'object' && args.shift()
-    const outgoing = {
```
Comment (paulmelnikow) [inline lib/icedfrisby.js]: This code was moved into `_makeRequest()` below, with a few changes.

**Comment by Fishbowler** [comment]
I noticed that too. Those 4 params were reverse engineered from having read the code and deduced (incorrectly) what was actually used. In reality, this is a set of options for `request` plus some extra bits of our own, no?

**Comment by Fishbowler** [comment]
This all looks like sensible replumbing. I've checked it out in place of the published npm package, and all of my tests act as I'd expect. There's some slightly different behaviour in failing tests when all run together, but that's looking to be the peculiarities of my repo and caused by #93 (in this case, different tests failing) and hopefully #106 will help to resolve it.

Unless you have any concerns, I'm happy to merge.

**Comment by paulmelnikow** [comment]
Yea, sounds great. Let's discuss the `params` issue further at some point.

**Pull request merged**